rust/datafusion/src/execution/context.rs

```diff
@@ -390,6 +391,37 @@ impl ExecutionContext {
         Ok(())
     }
 
+    /// Execute a query and write the results to a partitioned Parquet file
+    pub async fn write_parquet(
```

Comment:
eventually it would be cool to write this out with some pattern that is more controllable by the user